src/transformers/models/vision_text_dual_encoder/processing_vision_text_dual_encoder.py

```diff
@@ -0,0 +1,181 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for VisionTextDualEncoder
+"""
+from ...feature_extraction_utils import FeatureExtractionMixin
+from ...tokenization_utils import PreTrainedTokenizer
+from ...tokenization_utils_base import BatchEncoding
+from ...tokenization_utils_fast import PreTrainedTokenizerFast
+from ..auto.feature_extraction_auto import AutoFeatureExtractor
+from ..auto.tokenization_auto import AutoTokenizer
+
+
+class VisionTextDualEncoderProcessor:
+    r"""
+    Constructs a VisionTextDualEncoder processor which wraps a vision feature extractor and a tokenizer into a single
+    processor.
+
+    :class:`~transformers.VisionTextDualEncoderProcessor` offers all the functionalities of
+    :class:`~transformers.AutoFeatureExtractor` and :class:`~transformers.AutoTokenizer`. See the
+    :meth:`~transformers.VisionTextDualEncoderProcessor.__call__` and
+    :meth:`~transformers.VisionTextDualEncoderProcessor.decode` for more information.
+
+    Args:
+        feature_extractor (:class:`~transformers.AutoFeatureExtractor`):
+            The feature extractor is a required input.
+        tokenizer (:class:`~transformers.PreTrainedTokenizer`):
+            The tokenizer is a required input.
+    """
+
+    def __init__(self, feature_extractor, tokenizer):
+        if not isinstance(feature_extractor, FeatureExtractionMixin):
```

Comment:
```suggestion
        if not isinstance(feature_extractor, ImageFeatureExtractionMixin):
```

has to be the `ImageFeatureExtractionMixin`, no?

Reply:
that will introduce a direct dependency on `PIL` here,  and we will need to update the imports under 
`is_vision_available`. But we don't do this for other processors, so FeatureExtractionMixin should be fine IMO.